engine/src/org/pentaho/di/www/TransformationMap.java

```diff
@@ -391,16 +375,8 @@ public void setSlaveServerConfig( SlaveServerConfig slaveServerConfig ) {
   /**
    * @return the hostServerSocketPortsMap
    */
-  public Map<String, List<SocketPortAllocation>> getHostServerSocketPortsMap() {
-    return hostServerSocketPortsMap;
-  }
-
-  /**
-   * @param hostServerSocketPortsMap
-   *          the hostServerSocketPortsMap to set
-   */
-  public void setHostServerSocketPortsMap( Map<String, List<SocketPortAllocation>> hostServerSocketPortsMap ) {
-    this.hostServerSocketPortsMap = hostServerSocketPortsMap;
+  public List<SocketPortAllocation> getHostServerSocketPorts( String hostname ) {
+    return Collections.unmodifiableList( hostServerSocketPortsMap.get( hostname ) );
```

Comment:
@mbatchelor, Could you please explain why we need to create new ArrayList from existing one? Anyway it just copies references from the list in the constructor .